### PR TITLE
Ab#79731 implemt api

### PR DESCRIPTION
**Pull request by AlexandreSanscartier**

### CODE REVIEW

File: DocumentService.Unit.Tests/Controllers/DocumentsControllerTests.cs [hunk 1]
```diff
@@ -27,69 +28,6 @@ public DocumentsControllerTests()
             this.configuration = new Mock<IConfiguration>();
         }
 
-        //[Fact]
```
Comment (tc-chris) [inline DocumentService.Unit.Tests/Controllers/DocumentsControllerTests.cs]: Why did we remove this?

File: DocumentService/Controllers/DocumentsController.cs [hunk 2]
```diff
@@ -107,9 +107,28 @@ public IActionResult GetAllSpecifiedDocuments(string idString)
...
+                FileName = fileName,
+                Description = shortDescription,
+                SubmissionMethod = submissionMethod,
+                Language = fileLanguage,
+            };
+
+            try
+            {
+                var isUpdated = this.documentRepository.Update(documentInfo).Result;
+                return new JsonResult(isUpdated);
```
Comment (tc-chris) [inline DocumentService/Controllers/DocumentsController.cs:126]: Wouldn't return Ok(new { isUpdated }) be better

File: DocumentService/Controllers/DocumentsController.cs [hunk 3]
```diff
@@ -146,13 +169,15 @@ public IActionResult DeleteDocumentById(Guid id, string userName)
                 if (isDeleted)
                 {
                     return new JsonResult(isDeleted);
```
Comment (tc-chris) [inline DocumentService/Controllers/DocumentsController.cs:171]: Same thing here too, return Ok instead of new JsonResult

**Review by tc-chris — CHANGES_REQUESTED** [review]

File: DocumentService.Unit.Tests/Controllers/DocumentsControllerTests.cs [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+        }
+
+        [Fact]
+        public void GetDocumentById_True_WhenIdExists()
+        {
+            // Arrange
+            var guid = Guid.NewGuid();
+            var documentController = new DocumentsController(this.documentRepository.Object, this.azureBlobService.Object, this.configuration.Object);
+
+            DocumentInfo documentInfo = new DocumentInfo
```
Comment (TC-Cory) [inline DocumentService.Unit.Tests/Controllers/DocumentsControllerTests.cs:38]:
```suggestion
            Document documentInfo = new Document
```
Suggesting to use the DTO in this test

File: DocumentService.Unit.Tests/Controllers/DocumentsControllerTests.cs [hunk 5]
```diff
@@ -0,0 +1,182 @@
...
+            {
+                DocumentId = guid,
+                DateCreated = DateTime.UtcNow,
+                Description = "Generic Description",
+                FileName = "Test Doc",
+                DocumentTypes = new DocumentTypes { DocumentType = "Test", DocumentTypesId = 0 },
+                IsDeleted = false
+            };
```
Comment (TC-Cory) [inline DocumentService.Unit.Tests/Controllers/DocumentsControllerTests.cs:46]:
```suggestion
            };
              var dto = new DocumentDTO
            {
                Documents = new List<Document> { documentInfo }
            };
```

File: DocumentService.Unit.Tests/Controllers/DocumentsControllerTests.cs [hunk 6]
```diff
@@ -0,0 +1,182 @@
...
+            documentRepository.Setup(x => x.GetDocumentAsync(guid)).Returns(Task.FromResult(documentInfo));
```
Comment (TC-Cory) [inline DocumentService.Unit.Tests/Controllers/DocumentsControllerTests.cs:47]:
```suggestion
            documentRepository.Setup(x => x.GetDocumentAsync(guid)).Returns(Task.FromResult(dto));
```

File: DocumentService.Unit.Tests/Controllers/DocumentsControllerTests.cs [hunk 7]
```diff
@@ -0,0 +1,182 @@
...
+
+            // Act
+            var response = documentController.GetDocumentById(guid);
+            var res = response as OkObjectResult;
+            dynamic result = res.Value;
+            var docInfo = (DocumentInfo)result.GetType().GetProperty("document").GetValue(result, null);
```
Comment (TC-Cory) [inline DocumentService.Unit.Tests/Controllers/DocumentsControllerTests.cs:53]:
```suggestion
            var docInfo = (DocumentDTO)result.GetType().GetProperty("document").GetValue(result, null);
```
I suggest we cast the result to a DocumentDTO

File: DocumentService.Unit.Tests/Controllers/DocumentsControllerTests.cs [hunk 8]
```diff
@@ -0,0 +1,182 @@
...
+
+           // Assert
+            Assert.Equal(docInfo.FileName, documentInfo.FileName);
```
Comment (TC-Cory) [inline DocumentService.Unit.Tests/Controllers/DocumentsControllerTests.cs:56]:
```suggestion
            Assert.Equal(docInfo.Documents.First().FileName, documentInfo.FileName);
```

**Review by TC-Cory — COMMENTED** [review]
All looks good, just a small thing to change

File: DocumentService/Controllers/DocumentsController.cs (hunk 2 above)
Comment (AlexandreSanscartier) [reply DocumentService/Controllers/DocumentsController.cs:126]: JsonResult does the same thing as OK, but it formats the response as a JSONObject.

File: DocumentService/Controllers/DocumentsController.cs (hunk 3 above)
Comment (AlexandreSanscartier) [reply DocumentService/Controllers/DocumentsController.cs:171]: Same thing as the above comment.

File: DocumentService/Controllers/DocumentsController.cs (hunk 2 above)
Comment (tc-chris) [reply DocumentService/Controllers/DocumentsController.cs:138]: Fair enough

**Review by tc-chris — APPROVED** [review]
Looks good, nice work

**Pull request merged**